rayon-core/src/registry.rs

```diff
@@ -403,11 +412,27 @@ impl WorkerThread {
         self.registry.sleep.tickle(self.index);
     }
 
-    /// Pop `job` from top of stack, returning `false` if it has been
-    /// stolen.
     #[inline]
-    pub unsafe fn pop(&self) -> Option<JobRef> {
-        self.worker.pop()
+    pub fn local_deque_is_empty(&self) -> bool {
+        self.worker.len() == 0
+    }
+
+    #[inline]
+    pub fn breadth_first(&self) -> bool {
+        self.breadth_first
+    }
+
+    /// Attempts to obtain a "local" job -- typically this means
+    /// popping from the top of the stack, though if we are configured
+    /// for breadth-first execution, it would mean dequeuing from the
+    /// bottom.
+    #[inline]
+    pub unsafe fn take_local_job(&self) -> Option<JobRef> {
+        if !self.breadth_first {
+            self.worker.pop()
+        } else {
+            self.stealer.steal()
```

Comment:
I've just published a new version of Coco that introduces `Worker::steal` method. Now you can call `self.worker.steal()` here. This change will give some performance wins in breadth-first mode. I'd be curious to know how much :)